client/layout/masterbar/notifications.jsx

```diff
@@ -14,24 +14,28 @@ import Notifications from 'notifications';
 import store from 'store';
 import { recordTracksEvent } from 'state/analytics/actions';
 
-const MasterbarItemNotifications = React.createClass( {
-	propTypes: {
-		user: React.PropTypes.object.isRequired,
-		isActive: React.PropTypes.bool,
-		className: React.PropTypes.string,
-		onClick: React.PropTypes.func,
-		tooltip: React.PropTypes.string,
-	},
+class MasterbarItemNotifications extends Component {
+	constructor() {
+		super();
 
-	getInitialState() {
-		const user = this.props.user.get();
+		this.checkToggleNotes = this.checkToggleNotes.bind( this );
+		this.toggleNotesFrame = this.toggleNotesFrame.bind( this );
+		this.getNotificationLinkDomNode = this.getNotificationLinkDomNode.bind( this );
+		this.setNotesIndicator = this.setNotesIndicator.bind( this );
 
-		return {
+		this.state = {
 			isShowingPopover: false,
-			newNote: user && user.has_unseen_notes,
 			animationState: 0,
 		};
-	},
+	}
+
```

Comment:
Here too we could drop the constructor if the binds are replaced with arrow functions and the state becomes a class property:

```es6
class MasterbarItemNotifications extends Component {
  this.state = {
    isShowingPopover: false,
    ...
  };

  checkToggleNotes = () => { ... }
  ...
}
```